internal/utils/slices/generics.go

```diff
@@ -0,0 +1,12 @@
+package slices
+
+import "io"
+
+// TODO: consider using generics (if possible) to rewrite this
+func ToReaders(xs []io.ReadCloser) []io.Reader {
+	ys := make([]io.Reader, 0, len(xs))
+	for _, x := range xs {
+		ys = append(ys, x)
+	}
+	return ys
+}
```

Comment:
something like this: `func ToReaders[U ~[]T, T any](xs []U) []T {`, but I could not figure out how can `io.Reader` accept `io.ReadCloser` in append. Then it is not working in generics because we can not append type U to type T.

Reply:
I'm not sure if Go supports [Covariance and contravariance](https://en.wikipedia.org/wiki/Covariance_and_contravariance_(computer_science)).
I've tried several times to learn the topic but I systemically keep forgetting the theory behind it as it's not something that I use everyday.
You can have a look at it if you are **REALLY** curious but beware it is going to be neither easy nor quick 😄